lib/NodeFactory/NodeFactory.php

```diff
@@ -45,6 +46,9 @@ public function createDocumentNode(Environment $environment): DocumentNode;
      */
     public function createTocNode(Environment $environment, array $files, array $options): TocNode;
 
+    /** @param string[] $options */
+    public function createContentsNode(Environment $environment, DocumentNode $documentNode, array $options): ContentsNode;
```

Comment:
This is technically a BC-break. A solution would be to document this with `@method` on the interface, then switch from `@method` to a real method signature on 0.6.x, or to target 0.6.x directly. I think the latter solution would work be best, it's more straightforward and we don't need that level of user-friendliness for that library.

Reply:
Yeah there's only one implementation, and having that big of an interface makes it near impossible to implement… the reasoning behind it is here: https://github.com/doctrine/rst-parser/pull/13#discussion_r221410489

@jwage , I see you have a use case for this, do you actually have a custom implementation of this? And if you do use wrapping, do you have a proxy method for each and every method in this interface?